src/Symfony/Component/Yaml/Inline.php

```diff
@@ -104,6 +108,18 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
 
                 return 'null';
             case is_object($value):
+                if ($value instanceof \DateTime) {
+                    if ($value->getTimezone()->getName() === date_default_timezone_get()) {
+                        if (0 === (int) $value->format('His')) {
```

Comment:
Not sure if we really should cast here. Why not doing `'000000' === $value->format('His')` instead?